### PR TITLE
Amp+smog

### PR DESCRIPTION
**Pull request by vict0rsch**
Add systematic smog log if `d` is in the tasks. Need to add option for sky mask if `s` is also in the tasks

Images: input, smogged, target, prediction

⚠️ Always run `python test_trainer.py` before making a PR! (on the cluster, any GPU should work so just request an unkillable titan xp and testing will take you 3 minutes in total)

Add `--no_delete` to keep the comet exp and get this kind of images (resolution is 256 to speed things up, not the default 640)

val_r_d_4_#0_step_4

### CODE REVIEW

File: omnigan/trainer.py [hunk 1]
```diff
@@ -1614,3 +1618,43 @@ def del_output_path(self, force=False):
 
         if (Path(self.opts.output_path) / "is_functional.test").exists() or force:
             shutil.rmtree(self.opts.output_path)
+
+    def compute_smog(self, x, z=None, d=None, s=None, use_sky_seg=False):
```
Comment (alexhernandezgarcia) [inline omnigan/trainer.py:1622]: I struggle a bit with understanding the role of variables with single-letter names (z, d, s, A, I...)

File: omnigan/trainer.py [hunk 2]
```diff
@@ -1614,3 +1618,43 @@ def del_output_path(self, force=False):
...
+        A = airlight * torch.ones(3).view(1, -1, 1, 1)
+        I = srgb2lrgb(x)
+        vr = 1
+        beta_param = 2
+
+        beta = torch.tensor([beta_param / vr, beta_param / vr, beta_param / vr]).view(
+            1, -1, 1, 1
+        )
+        d = normalize(d, mini=0.3, maxi=1.0)
+        d = 1.0 / d
```
Comment (alexhernandezgarcia) [inline omnigan/trainer.py:1652]: What is this inversion for?

File: omnigan/trainer.py (hunk 1 above)
Comment (vict0rsch) [reply omnigan/trainer.py:1622]: I understand it can get a little confusing. All over the code, in general:

* `m` refers to "mask", either as input (`batch["data"]["m"]`) or output `G.decoders["m"]` or as a mask variable with shape `batch x 1 x height x width`
* `d` refers similarly to "depth"
* `s` to "segmentation" when it is a task or "simulated" when it is a domain
* `z` always refers to the latent vector, *i.e.* the output of the encoder, *i.e.* the input to the decoders
  * `z` is a tensor `batch x 2048 x 80 x 80` when the encoder is `deeplabv2`, it is a tuple (high_level_features, low_level_features) when the encoder is `deeplabv3`. High vs Low level features means we extract both low-level feature maps in the middle of the encoder and high-level feature maps as its output
  * in the case of dlv3 we may use the `mobilenet` or `resnet` backbones, one has fewer parameters than the other

In the special case of this function the other single letter vars are not standard and we should actually rename them

Comment (vict0rsch) [reply omnigan/trainer.py:1622]: updated @alexhernandezgarcia :)

File: omnigan/trainer.py (hunk 2 above)
Comment (vict0rsch) [reply omnigan/trainer.py:1650]: Because the values stored are actually inverse depth, so that the sky needs to be predicted at 0 and not +inf and there is no inf problem because no object has a depth of 0 (which could theoretically yield +inf)
